File: circle.java
/**
 * The Circle class models a circle with a radius and color.
 */
public class Circle {    // Save as "Circle.java"
   // Private instance variables
   private double radius;
   private String color;

   // Constructors (overloaded)
   /** Constructs a Circle instance with default radius and color */
   public Circle() {                   // 1st Constructor (default constructor)
      radius = 1.0;
      color = "red";
   }
   /** Constructs a Circle instance with the given radius and default color*/
   public Circle(double r) {           // 2nd Constructor
      radius = r;
      color = "red";
   }
   /** Constructs a Circle instance with the given radius and color */
   public Circle(double r, String c) { // 3rd Constructor
      radius = r;
      color = c;
   }

   // Public methods
   /** Returns the radius */
   public double getRadius() {  // getter for radius
      return radius;
   }
   /** Returns the color */
   public String getColor() {   // getter for color
      return color;
   }
   /** Returns the area of this circle */
   public double getArea() {
      return radius * radius * Math.PI;
   }
}

/**
 * A Test Driver for the "Circle" class
 */
public class TestCircle {    // Save as "TestCircle.java"
   public static void main(String[] args) {   // Program entry point
      // Declare and Construct an instance of the Circle class called c1
      Circle c1 = new Circle(2.0, "blue");  // Use 3rd constructor
      System.out.println("The radius is: " + c1.getRadius());  // use dot operator to invoke member methods
      //The radius is: 2.0
      System.out.println("The color is: " + c1.getColor());
      //The color is: blue
      System.out.printf("The area is: %.2f%n", c1.getArea());
      //The area is: 12.57

      // Declare and Construct another instance of the Circle class called c2
      Circle c2 = new Circle(2.0);  // Use 2nd constructor
      System.out.println("The radius is: " + c2.getRadius());
      //The radius is: 2.0
      System.out.println("The color is: " + c2.getColor());
      //The color is: red
      System.out.printf("The area is: %.2f%n", c2.getArea());
      //The area is: 12.57

      // Declare and Construct yet another instance of the Circle class called c3
      Circle c3 = new Circle();  // Use 1st constructor
      System.out.println("The radius is: " + c3.getRadius());
      //The radius is: 1.0
      System.out.println("The color is: " + c3.getColor());
      //The color is: red
      System.out.printf("The area is: %.2f%n", c3.getArea());
      //The area is: 3.14
   }
}

File: circle.cpp
#include <iostream>
#include <string>
#include <cmath>

using namespace std;

class Circle {
public:
  // The public constants
  static const double DEFAULT_RADIUS = 1.0;
  static const string DEFAULT_COLOR = "red";

  // The private instance variables
  double radius;
  string color;

  // The (overloaded) constructors
  /** Constructs a Circle with default radius and color */
  Circle() {  // 1st (default) Constructor
    this->radius = DEFAULT_RADIUS;
    this->color = DEFAULT_COLOR;
  }
  /** Constructs a Circle with the given radius and default color */
  Circle(double radius) {  // 2nd Constructor
    this->radius = radius;
    this->color = DEFAULT_COLOR;
  }
  /** Constructs a Circle with the given radius and color */
  Circle(double radius, string color) {  // 3rd Constructor
    this->radius = radius;
    this->color = color;
  }

  /** Returns the radius - the public getter for private variable radius. */
  double getRadius() {
    return this->radius;
  }
  /** Sets the radius - the public setter for private variable radius */
  void setRadius(double radius) {
    this->radius = radius;
  }
  /** Returns the color - the public getter for private variable color */
  string getColor() {
    return this->color;
  }
  /** Sets the color - the public setter for private variable color */
  void setColor(string color) {
    this->color = color;
  }

  /** Returns a self-descriptive string for this Circle instance */
  string toString() {
    return "Circle[radius=" + to_string(radius) + ", color=" + color + "]";
  }

  /** Returns the area of this Circle */
  double getArea() {
    return radius * radius * M_PI;
  }

  /** Returns the circumference of this Circle */
  double getCircumference() {
    return 2.0 * radius * M_PI;
  }
};

int main() {
  // Create a Circle object with default radius and color
  Circle circle1;
